packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -29,11 +29,115 @@ export type TestOperation = (
 	random: IRandom,
 ) => IMergeTreeOp | undefined;
 
+export const insertField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const chunkLength = random.integer(1, 10);
+	const numberText: string = (client.longClientId!.codePointAt(0)! % 10)
+		.toString()
+		.repeat(chunkLength);
+	if (getFieldEndpoints(client, opStart, opEnd) === undefined) {
+		return client.insertTextLocal(opStart, `{${numberText}}`);
+	}
+};
+
+const posInField = (
+	client: TestClient,
+	pos: number,
+): { startPos: number; endPos: number } | undefined => {
+	if (
+		pos >= client.getLength() ||
+		(!Number.isInteger(Number(client.getText(pos, pos + 1))) &&
+			client.getText(pos, pos + 1) !== "{" &&
+			client.getText(pos, pos + 1) !== "}")
+	) {
+		return undefined;
+	}
+
+	let startPos = pos;
+	let endPos = pos;
+	// To find the start and end separators, walk backwards and forwards until the desired character is found.
+	while (
+		startPos > 0 &&
+		client.getText(startPos, startPos + 1) !== "{" &&
+		(client.getText(startPos, startPos + 1) === "}" ||
+			Number.isInteger(Number(client.getText(startPos, startPos + 1))))
+	) {
+		startPos--;
+	}
+
+	while (
+		endPos < client.getLength() &&
+		client.getText(endPos, endPos + 1) !== "}" &&
+		(client.getText(endPos, endPos + 1) === "{" ||
+			Number.isInteger(Number(client.getText(endPos, endPos + 1))))
+	) {
+		endPos++;
+	}
+
+	return { startPos, endPos };
+};
+
+const getFieldEndpoints = (
+	client: TestClient,
+	start: number,
+	end: number,
+): { startPos: number; endPos: number } | undefined => {
+	const startField = posInField(client, start);
+	const endField = posInField(client, end);
+
+	if (startField === undefined && endField === undefined) {
+		return undefined;
+	}
+	return startField ?? endField;
+};
+
+export const obliterateField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const fieldEndpoints = getFieldEndpoints(client, opStart, opEnd);
+
+	let endISP: InteriorSequencePlace | undefined;
+	if (fieldEndpoints !== undefined) {
+		const { startPos, endPos } = fieldEndpoints;
+		if (endPos >= client.getLength()) {
+			endISP = { pos: client.getLength() - 1, side: Side.After };
+		}
+		// Concurrent to the obliterate, replace the same range with new text.
+		insertField(client, startPos, endPos, random);
```

Comment:
To simulate the case we care about, preserve the "{" & "}" rather than obliterating and re-adding. Just obliterate and replace the content in between.